File: dataset/output/PERPLEXITY_llama-3-sonar-large-32k-chat/Task30_PERPLEXITY_llama-3-sonar-large-32k-chat.java
public class Main {
    public static void main(String[] args) {
        System.out.println(longest("xyaabbbccccdefww", "xxxxyyyyabklmopq"));
        System.out.println(longest("abcdefghijklmnopqrstuvwxyz", "abcdefghijklmnopqrstuvwxyz"));
    }

    public static String longest(String s1, String s2) {
        String combined = s1 + s2;
        String unique = "";
        for (char c : combined.toCharArray()) {
            if (!unique.contains(String.valueOf(c))) {
                unique += c;
            }
        }
        char[] chars = unique.toCharArray();
        java.util.Arrays.sort(chars);
        return new String(chars);
    }
}

File: dataset/output/PERPLEXITY_llama-3-sonar-large-32k-chat/Task30_PERPLEXITY_llama-3-sonar-large-32k-chat.cpp
#include <iostream>
#include <set>
#include <string>
#include <algorithm>

std::string longest(const std::string& a, const std::string& b) {
    std::set<char> set;
    for (char c : a) set.insert(c);
    for (char c : b) set.insert(c);
    std::string result;
    for (char c : set) result += c;
    std::sort(result.begin(), result.end());
    return result;
}

int main() {
    std::cout << longest("xyaabbbccccdefww", "xxxxyyyyabklmopq") << std::endl;
    std::cout << longest("abcdefghijklmnopqrstuvwxyz", "abcdefghijklmnopqrstuvwxyz") << std::endl;
    return 0;
}